nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-5-processors/src/main/java/org/apache/nifi/processors/elasticsearch/JsonQueryElasticsearch5.java

```diff
@@ -0,0 +1,291 @@
+        descriptors.add(BREAK_UP_HITS);
+        descriptors.add(BREAK_UP_AGGREGATIONS);
+
+        propertyDescriptors = Collections.unmodifiableList(descriptors);
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return relationships;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return propertyDescriptors;
+    }
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+        clientService = context.getProperty(CLIENT_SERVICE).asControllerService(ElasticSearchClientService.class);
+    }
+
+    @OnUnscheduled
+    public void onUnscheduled() {
+        this.clientService = null;
+    }
+
+
+    private ObjectMapper mapper = new ObjectMapper();
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        try {
+
+            String query = context.getProperty(QUERY).evaluateAttributeExpressions(flowFile).getValue();
+            String index = context.getProperty(INDEX).evaluateAttributeExpressions(flowFile).getValue();
+            String type = context.getProperty(TYPE).evaluateAttributeExpressions(flowFile).getValue();
+
+            Optional<ElasticSearchResponse> resp = clientService.search(query, index, type);
+            ElasticSearchResponse response = resp.get();
+
+            List<FlowFile> hitsFlowFiles = handleHits(response.getHits(), context, session, flowFile);
+            List<FlowFile> aggsFlowFiles = handleAggregations(response.getAggregations(), context, session, flowFile);
+
+            if (hitsFlowFiles.size() > 0) {
+                session.transfer(hitsFlowFiles, REL_HITS);
+                for (FlowFile ff : hitsFlowFiles) {
+                    session.getProvenanceReporter().create(ff);
+                }
+            }
+
+            if (aggsFlowFiles.size() > 0) {
+                session.transfer(aggsFlowFiles, REL_AGGREGATIONS);
+                for (FlowFile ff : aggsFlowFiles) {
+                    session.getProvenanceReporter().create(ff);
+                }
+            }
+            session.transfer(flowFile, REL_SUCCESS);
+        } catch (Exception ex) {
+            getLogger().error("Error processing flowfile.", ex);
+            session.transfer(flowFile, REL_FAILURE);
+        }
+    }
+
+    private FlowFile writeAggregationFlowFileContents(String name, String json, ProcessSession session, FlowFile aggFlowFile) {
+        aggFlowFile = session.write(aggFlowFile, new OutputStreamCallback() {
+            @Override
+            public void process(OutputStream out) throws IOException {
+                out.write(json.getBytes());
+            }
+        });
+        if (name != null) {
+            aggFlowFile = session.putAttribute(aggFlowFile, "aggregation.name", name);
```

Comment:
Recommend adding a @WritesAttributes annotation to the processor class containing any attributes like aggregation.name, mime.type, etc. that may be added/updated by this processor